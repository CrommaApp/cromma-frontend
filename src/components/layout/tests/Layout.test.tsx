import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { RecoilRoot } from 'recoil';
import Layout from '@components/layout';

jest.mock('@hooks/useErrorMessage', () => ({
	__esModule: true,
	default: () => '다시 시도해주세요.',
}));

jest.mock('@hooks/useSuccessMessage', () => ({
	__esModule: true,
	default: () => '로그인 성공',
}));

describe('Layout Component', () => {
	beforeEach(() => {
		render(
			<RecoilRoot>
				<BrowserRouter>
					<Layout>
						<></>
					</Layout>
				</BrowserRouter>
			</RecoilRoot>,
		);
	});

	it('Snapshot', () => {
		const component = renderer.create(
			<RecoilRoot>
				<BrowserRouter>
					<Layout>
						<></>
					</Layout>
				</BrowserRouter>
			</RecoilRoot>,
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
