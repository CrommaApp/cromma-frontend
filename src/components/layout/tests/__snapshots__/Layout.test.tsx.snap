// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Snapshot 1`] = `
<div
  aria-hidden={false}
  className="sc-eCImPb jVnAoT"
>
  <div
    className="sc-jRQBWg jUpBZr"
    role="alert"
  >
    <p>
      다시 시도해주세요.
    </p>
  </div>
  <div
    className="sc-jRQBWg jUpBZr"
    role="status"
  >
    <p>
      로그인 성공
    </p>
  </div>
  <div
    className="sc-gKclnd crsHSz"
  >
    <div>
      <p
        className="sc-iCfMLu dUIAQX"
      >
        로그인 해주세요
      </p>
      <nav
        className="sc-furwcr jhiwlJ"
      >
        <ul>
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              홈
            </a>
          </li>
          <li>
            <a
              href="/post/upload"
              onClick={[Function]}
            >
              게시글 작성
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      회원가입/로그인
    </button>
  </div>
  <main />
</div>
`;
